ext/dom/tests/DOMDocument_importNode_attribute_prefix_conflict.phpt

```diff
@@ -0,0 +1,65 @@
+--TEST--
+DOMDocument::importNode() with attribute prefix name conflict
+--EXTENSIONS--
+dom
+--FILE--
+<?php
+
+echo "--- Non-default namespace test case without a default namespace in the destination ---\n";
+
+$dom1 = new DOMDocument();
+$dom2 = new DOMDocument();
+$dom1->loadXML('<?xml version="1.0"?><container xmlns:foo="http://php.net" foo:bar="yes"/>');
+$dom2->loadXML('<?xml version="1.0"?><container xmlns:foo="http://php.net/2"/>');
+$attribute = $dom1->documentElement->getAttributeNode('foo:bar');
+$imported = $dom2->importNode($attribute);
+$dom2->documentElement->setAttributeNodeNS($imported);
+
+echo $dom1->saveXML();
+echo $dom2->saveXML();
+
+echo "--- Non-default namespace test case with a default namespace in the destination ---\n";
+
+$dom1 = new DOMDocument();
+$dom2 = new DOMDocument();
+$dom1->loadXML('<?xml version="1.0"?><container xmlns:foo="http://php.net" foo:bar="yes"/>');
+$dom2->loadXML('<?xml version="1.0"?><container xmlns="http://php.net" xmlns:foo="http://php.net/2"/>');
+$attribute = $dom1->documentElement->getAttributeNode('foo:bar');
+$imported = $dom2->importNode($attribute);
+$dom2->documentElement->setAttributeNodeNS($imported);
+
+echo $dom1->saveXML();
+echo $dom2->saveXML();
+
+echo "--- Default namespace test case ---\n";
+
+// We don't expect the namespace to be imported because default namespaces on the same element don't apply to attributes
+// but the attribute should be imported
+$dom1 = new DOMDocument();
+$dom2 = new DOMDocument();
+$dom1->loadXML('<?xml version="1.0"?><container xmlns="http://php.net" bar="yes"/>');
+$dom2->loadXML('<?xml version="1.0"?><container xmlns="http://php.net/2"/>');
+$attribute = $dom1->documentElement->getAttributeNode('bar');
+$imported = $dom2->importNode($attribute);
+$dom2->documentElement->setAttributeNodeNS($imported);
+
+echo $dom1->saveXML();
+echo $dom2->saveXML();
+
+?>
+--EXPECT--
+--- Non-default namespace test case without a default namespace in the destination ---
+<?xml version="1.0"?>
+<container xmlns:foo="http://php.net" foo:bar="yes"/>
+<?xml version="1.0"?>
+<container xmlns:foo="http://php.net/2" xmlns:default="http://php.net" default:bar="yes"/>
+--- Non-default namespace test case with a default namespace in the destination ---
+<?xml version="1.0"?>
+<container xmlns:foo="http://php.net" foo:bar="yes"/>
+<?xml version="1.0"?>
+<container xmlns="http://php.net" xmlns:foo="http://php.net/2" xmlns:default="http://php.net" default:bar="yes"/>
```

Comment:
While technically correct, shouldn't/couldn't the `default:`-prefix be avoided, given `container` is in the same namespace?

Reply:
This even feels counter intuitive, given the next test actually even ignores the namespace.

Reply:
Default namespaces don't apply to attributes, so we need the explicit default: prefix.

Reply:
That's also why the next test ignores the default namespace.

Reply:
Sorry, I don't understand what you mean.

If we write `bar` instead of `default:bar`, then the bar attribute belongs to no namespace. Example from Javascript:
```js
dom = (new DOMParser).parseFromString('<?xml version="1.0"?><container xmlns="http://php.net" bar="yes"/>', 'text/xml')
dom.documentElement.attributes[1].namespaceURI // outputs: null
```

versus:
```js
dom = (new DOMParser).parseFromString('<?xml version="1.0"?><container xmlns="http://php.net" xmlns:default="http://php.net" default:bar="yes"/>', 'text/xml')
dom.documentElement.attributes[2].namespaceURI // outputs: http://php.net
```

Reply:
You're right. Disregard. I should probably go sleep :)